pkg/ext/resources/tokens/types.go

```diff
@@ -0,0 +1,44 @@
+package tokens
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RancherToken
+type RancherToken struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the spec of RancherToken
+	Spec RancherTokenSpec `json:"spec"`
+	// +optional
+	Status RancherTokenStatus `json:"status"`
+}
+
+type RancherTokenSpec struct {
+	// UserID is the user id
+	UserID string `json:"userID"`
```

Comment:
Is that principal id of the user, or the name kube name of a User resource ?
Which would be better ?
The RFC currently takes a principal id and stores a user resource name
(which means conversions on in- and output :( - store both? go for resource name ?)

Answers go into the RFC.